lib/ansible/module_utils/basic.py

```diff
@@ -242,6 +243,8 @@
     delimiter=dict(),  # used by assemble
     directory_mode=dict(),  # used by copy
     unsafe_writes=dict(type='bool'),  # should be available to any module using atomic_move
+    lock=dict(type='bool', default=False),
```

Comment:
We probably do not want to add new arguments into file_common_args unless they are about setting permissions.

Reply:
file itself wont use them, they can be in their own vars LOCK_ARGs for file modules that want to use them 

We actually want to move away from most of these that are 'module specific' and leave backup/unsafe_writes/etc other generics in a FILE_MODIFY_ARGS or some other structure

Reply:
Sorry not sure if I understand correctly. So I can just create a structure somewhere after FILE_COMMON_ARGUMENTS and put my stuff there?